software/firmware/europi.py

```diff
@@ -268,46 +290,78 @@ class DigitalInput(DigitalReader):
     ``x.handler(new_function)``. Do not include the brackets for the function,
     and replace the 'x' in the example with the name of your input, either
     ``b1``, ``b2``, or ``din``.
+
+    Here is another example how you can write digital input handlers to react
+    to a clock source and match its trigger duration.::
+
+        @din.handler
+        def gate_on():
+            # Trigger outputs with a probability set by knobs.
+            cv1.value(random() > k1.percent())
+            cv2.value(random() > k2.percent())
+
+        @din.handler_falling
+        def gate_off():
+            # Turn off all triggers on falling clock trigger to match clock.
+            cv1.off()
+            cv2.off()
+
+    When writing a handler, try to keep the code as minimal as possible.
+    Ideally handlers should be used to change state and allow your main loop
+    to change behavior based on the altered state. See `tips <https://docs.micropython.org/en/latest/reference/isr_rules.html#tips-and-recommended-practices>`_
+    from the MicroPython documentation for more details.

```

Comment:
This addition and some of the other changes are not showing up in your published docs, and I can't figure out why.